sdk/src/main/scala/io/horizen/history/AbstractHistory.scala

```diff
@@ -368,20 +388,14 @@ abstract class AbstractHistory[
         else
           Younger
       case _ =>
-        val otherBestKnownBlockIndex = dSuffix.size - 1 - dSuffix.reverse.indexWhere(id => storage.heightOf(id).isDefined)
-        val otherBestKnownBlockHeight = storage.heightOf(dSuffix(otherBestKnownBlockIndex)).get
-        // other node height can be approximatly calculated as height of other KNOWN best block height + size of rest unknown blocks after it.
-        // why approximately? see knownBlocksHeightToSync algorithm: blocks to sync step increasing.
-        // to do: need to discuss
-        val otherBestBlockApproxHeight = otherBestKnownBlockHeight + (dSuffix.size - 1 - otherBestKnownBlockIndex)
-        if (storage.height < otherBestBlockApproxHeight)
+        val otherBestKnownBlockHeight = storage.heightOf(dSuffix.head).get
+        val otherBestBlockApproxHeight = calculateOtherNodeApproxHeight(otherBestKnownBlockHeight, dSuffix.size)
+        if(otherBestBlockApproxHeight == -1)
+          Unknown
+        else if (storage.height < otherBestBlockApproxHeight)
           Older
-        else if (storage.height == otherBestBlockApproxHeight) {
-          if(otherBestBlockApproxHeight == otherBestKnownBlockHeight)
-            Equal // UPDATE: FORK in both cases
-          else
-            Fork
-        }
+        else if (storage.height == otherBestBlockApproxHeight)
+          Fork
         else
           Younger
```

Comment:
Do you think it is safe to set the status to Younger as default here?
This case means that that other node has fewer blocks, but there is also a Fork, because a number of blocks from the other node history was not found in local history.
So it is both Fork and Younger

Reply:
Good point, after further analysis I proceed to refactor the method logic.
The current logic in case the divergent suffix sequence size is greater than 1 is the following:
knowing our block height, the best known block height and the calculated best block related to the other node history, we do the following checks:
- if the other node best block height is greater than the current height:
	- and the height of the second index of the divergent suffix sequence is less then the current height we have a `Fork` (this because it means that the block id at a height less than our best block didn't found a match)
	- else the other node history is `Older`
the default condition for the other cases is `Fork`.

No need to manage the `Younger` case because in case the other node is really younger (with no forks) the divergent suffix sequence size will be 1 and we will be outside these cases

The `Unknown` condition is satisfied only if we are not able to calculate the other node best block height, in this case we don't have enough data to return a secure comparison result.